odl/discr/grid.py

```diff
@@ -327,12 +377,43 @@ def max(self, **kwargs):
         else:
             return self.max_pt
 
+    @property
+    def mid_pt(self):
+        """Midpoint of the grid, not necessarily a grid point.
+
+        Examples
+        --------
+        >>> rg = uniform_grid([-1.5, -1], [-0.5, 3], (2, 3))
+        >>> rg.mid_pt
+        array([-1.,  1.])
+        """
+        return (self.max_pt + self.min_pt) / 2
+
+    @property
+    def stride(self):
+        """Step per axis between neighboring points of a uniform grid.
+
+        Examples
+        --------
+        >>> rg = uniform_grid([-1.5, -1], [-0.5, 3], (2, 3))
+        >>> rg.stride
+        array([ 1.,  2.])
+        """
+        if not self.is_uniform:
+            raise NotImplementedError('`stride` not defined for non-uniform '
+                                      'grid')
+        strd = np.zeros(self.ndim)
```

Comment:
Okay, "shadow" was the wrong term, "simultaneously highlighted by the editor" would be more precise.